roles/kubernetes-apps/external_cloud_controller/openstack/tasks/main.yml

```diff
@@ -2,27 +2,16 @@
 - include_tasks: openstack-credential-check.yml
   tags: external-openstack
 
-- name: External OpenStack Cloud Controller | Write cacert file
-  include_tasks: openstack-write-cacert.yml
-  run_once: true
-  loop: "{{ groups['k8s_cluster'] }}"
-  loop_control:
-    loop_var: delegate_host_to_write_cacert
+- name: External OpenStack Cloud Controller | Get base64 cacert
+  slurp:
+    src: "{{ external_openstack_cacert }}"
+  register: external_openstack_cacert_b64
   when:
-    - inventory_hostname in groups['k8s_cluster']
+    - inventory_hostname == groups['k8s_control_plane'][0]
     - external_openstack_cacert is defined
     - external_openstack_cacert | length > 0
   tags: external-openstack
 
-- name: External OpenStack Cloud Controller | Write External OpenStack cloud-config
-  template:
-    src: "external-openstack-cloud-config.j2"
```

Comment:
Can't we remove this template file external-openstack-cloud-config.j2 with this pull request?

Reply:
I think this actually needs a bit of rethought now that you pointed it out. The `external-openstack-cloud-config` is rendered onto the first tube controller then it is slurp'ed up and put into the secret, so the answer to your question is no, we cannot remove that file, but I should have not deleted this task in the first place. I pushed a fix for this.

Reply:
@oomichi I modified the logic to avoid writing the `external-openstack-cloud-config` to any node and write it directly in the Kubernetes secret.